server/hedley/modules/custom/hedley_admin/scripts/generate-demographics-hc-report.php

```diff
@@ -11,13 +11,53 @@
 require_once __DIR__ . '/report_common.inc';
 
 $limit_date = drush_get_option('limit_date', FALSE);
-$region = drush_get_option('region', FALSE);
-
+$region_name = drush_get_option('region', FALSE);
 if (!$limit_date) {
   drush_print('Please specify --limit_date option');
   exit;
 }
 
+/**
+ * Gets the node ID of the health center.
+ *
+ * @param string $health_center_name
+ *   The name of a healthcenter.
+ *
+ * @return string
+ *   The node ID of the health center.
+ */
+function get_health_center_id($health_center_name = NULL) {
+  if ($health_center_name) {
+    $health_center_name = str_replace(['_', '-', '.'], ' ', $health_center_name);
+    $results = db_query("SELECT nid FROM node WHERE title='$health_center_name' AND type='health_center'");
+    if (!$results->fetchField()) {
+      $results = db_query("SELECT nid FROM node WHERE title LIKE '$health_center_name%' AND type='health_center'");
+      if (!$results->fetchField()) {
+        drush_print('No health centers match the name provided');
+        exit;
+      }
+      elseif (!$results->fetchField()) {
+        return db_query("SELECT nid FROM node WHERE title LIKE '$health_center_name%' AND type='health_center' LIMIT 1")->fetchField();
+      }
+      else {
+        $results = db_query("SELECT nid FROM node WHERE title LIKE '$health_center_name%' AND type='health_center'");
+        drush_print('Multiple health centers match the name provided including ' .
+          get_health_center($results->fetchField()) . ', ' . get_health_center($results->fetchField()) .
+          ", etc. \r\nPlease use a more specific name");
+        exit();
+      }
+    }
+    else {
+      return db_query("SELECT nid FROM node WHERE title='$health_center_name' AND type='health_center'")->fetchField();
+    }
+  }
+  else {
+    return NULL;
+  }
+}
+
+$region = get_health_center_id($region_name);
```

Comment:
This is confusing. Is it a region or a health center? We should somehow resolve this to have coherent naming.

Reply:
I see that this confusion already exist, but still, it would be nice to address it.

Reply:
@smericle You can rename this parameter from to original PR.